src/providers/network-connection-provider.test.tsx

```diff
@@ -0,0 +1,202 @@
+import React from "react";
+import { act, cleanup, render } from "@testing-library/react";
+import { NetworkConnectionProvider } from "./network-connection-provider";
+import {
+    NetworkConnection,
+    NetworkInformationUtils,
+} from "andculturecode-javascript-core";
+import { useNetworkConnection } from "../hooks/use-network-connection";
+import { networkConnectionFactory } from "../tests/factories/network-connection-factory";
```

Comment:
We generally don't pull in factories directly like this - can you register this statically in `src/tests/factories/index.ts` and reference it via `Factory` from `rosie`?